torch/lib/c10d/test/TCPStoreTest.cpp

```diff
@@ -77,7 +77,7 @@ void testHelper(const std::string& prefix = "") {
 
   for (auto i = 0; i < numThreads; i++) {
     threads.push_back(
-        std::thread([&sem1, &sem2, &clientStores, i, &expectedCounterRes] {
+        std::thread([&sem1, &sem2, &clientStores, i, &expectedCounterRes, &numIterations, &numThreads] {
```

Comment:
`numIterations` and `numThreads` can be read without being captured since they are const-initialized right? Was this causing any test failures?

Reply:
This will cause compile error on MSVC. So added them.